app/main.py

```diff
@@ -1,3 +1,66 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance | int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: Distance | int) -> Distance:
+        if isinstance(other, Distance):
+            raise TypeError(
+                "'__mul__' method should not accept Distance instance")
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Distance | int) -> Distance:
+        if isinstance(other, Distance):
+            raise TypeError(
+                "'__truediv__' method should not accept Distance instance")
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Distance | int) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        else:
+            return self.km < other
+
+    def __gt__(self, other: Distance | int) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        else:
+            return self.km > other
+
+    def __eq__(self, other: Distance | int) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        else:
+            return self.km == other
```

Comment:
```suggestion
        if isinstance(other, Distance):
            return self.km == other.km
        return self.km == other
```
`else` redundant here 
please fix in other places